crates/ra_ide/src/inlay_hints.rs

```diff
@@ -1052,4 +1098,124 @@ fn main() {
         "###
         );
     }
+
+    #[test]
+    fn chaining_hints_ignore_comments() {
+        let (analysis, file_id) = single_file(
+            r#"
+            struct A(B);
+            impl A { fn into_b(self) -> B { self.0 } }
+            struct B(C)
```

Comment:
One small thing: here and below, you've forgotten a semicolon, this is technically a compiler error.

Reply:
> It feels like we need another iteration on hints shortening:
Indeed, things do start to get a bit cluttered for complicated nested types.

`{unknown}` being so long and prevalent doesn't help.

> One small thing: here and below, you've forgotten a semicolon, this is technically a compiler error.

Will fix it.